testing/testing-utils/src/generators.rs

```diff
@@ -149,7 +149,9 @@ pub fn sign_tx_with_key_pair(key_pair: Keypair, tx: Transaction) -> TransactionS
     let signature =
         sign_message(B256::from_slice(&key_pair.secret_bytes()[..]), tx.signature_hash()).unwrap();
 
-    TransactionSigned::from_transaction_and_signature(tx, signature)
+    let tx = TransactionSigned::from_transaction_and_signature(tx, signature);
+    tx.hash_ref(); // initializes transaction hash
```

Comment:
same q

Reply:
wondering if it would make sense to delegate `Hash` and `Eq` to transaction without a hash? given that we now never fill the hash with wrong data

Reply:
I'd prefer manual PartialEq, here

I actually find that strange that OnceLock doesn't do this